service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/generator/UploadedOrderDocumentGenerator.java

```diff
@@ -21,14 +21,18 @@
     private final DocumentDownloadService documentDownloadService;
 
     public OrderDocumentGeneratorResult generate(CaseData caseData, OrderStatus status, RenderFormat format) {
+        return generate(caseData, status, format, needSealing(caseData, status));
+    }
 
+    public OrderDocumentGeneratorResult generate(CaseData caseData, OrderStatus status, RenderFormat format,
+                                                 boolean needSealing) {
         DocumentReference documentReference = caseData.getManageOrdersEventData().getManageOrdersUploadOrderFile();
```

Comment:
It will still fail with the new docstore. I refactored those code a bit. If user uploaded a document, FPL won't call this generate() method until end of the flow. It is pointless for FPL to re-upload the document uploaded by user, where the content of the document is exactly the same.

This method is triggered 1) when previewing the document or 2) at the end of the flow to seal the document. 
Before entering the"preview page" of "upload document" flow, FPL will download the document uploaded by user and re-upload the same document to docstore (p.s. convert it to PDF if necessary, but the content are still the same), then user can preview the document they uploaded at the preview page.

Reply:
Can we double check if we need this change?

Reply:
You are right, we don't need this change anymore.